quarkus-integration-test-vault-app is requiring Docker to build

**Describe the bug**
The module `quarkus-integration-test-vault-app` is expecting the docker daemon to be running; not having it has the whole Quarkus build fail.

Other modules have a conditional dependency controlled by a flag to enable any integration tests requiring docker.

This is particularly annoying for people building Quarkus without docker (possibly because working on an unrelated set of modules), or using `podman` instead of docker.

I wasn't setting that flag, but I had `-Dnative`; this will produce:

```
Caused by: java.lang.IllegalStateException: Could not find a valid Docker environment. Please see logs and check configuration
	at org.testcontainers.dockerclient.DockerClientProviderStrategy.lambda$getFirstValidStrategy$3(DockerClientProviderStrategy.java:156)
	at java.util.Optional.orElseThrow(Optional.java:290)
	at org.testcontainers.dockerclient.DockerClientProviderStrategy.getFirstValidStrategy(DockerClientProviderStrategy.java:148)
	at org.testcontainers.DockerClientFactory.client(DockerClientFactory.java:111)
	at org.testcontainers.containers.GenericContainer.<init>(GenericContainer.java:175)
	at org.testcontainers.containers.JdbcDatabaseContainer.<init>(JdbcDatabaseContainer.java:36)
	at org.testcontainers.containers.PostgreSQLContainer.<init>(PostgreSQLContainer.java:32)
	at org.testcontainers.containers.PostgreSQLContainer.<init>(PostgreSQLContainer.java:28)
	at io.quarkus.vault.test.VaultTestExtension.start(VaultTestExtension.java:120)
	at io.quarkus.vault.test.VaultTestLifecycleManager.start(VaultTestLifecycleManager.java:32)
	at io.quarkus.test.common.TestResourceManager.start(TestResourceManager.java:43)
```

--- a/integration-tests/vault-app/pom.xml
+++ b/integration-tests/vault-app/pom.xml
@@ -75,7 +75,6 @@
             <plugin>
                 <groupId>io.quarkus</groupId>
                 <artifactId>quarkus-maven-plugin</artifactId>
-                <version>${project.version}</version>
                 <executions>
                     <execution>
                         <goals>
@@ -109,6 +108,12 @@
                                 <javax.net.ssl.trustStore>${project.basedir}/src/test/resources/vaultTrustStore</javax.net.ssl.trustStore>
                                 <vault-test-extension.use-tls>false</vault-test-extension.use-tls>
                             </systemProperties>
+                        </configuration>
+                    </plugin>
+                    <plugin>
+                        <artifactId>maven-failsafe-plugin</artifactId>
+                        <configuration>
+                            <skip>false</skip>
                         </configuration>
                     </plugin>
                 </plugins>
@@ -149,12 +154,9 @@
                                 </configuration>
                             </execution>
                         </executions>
-                        <configuration>
-                            <skip>false</skip>
-                        </configuration>
                     </plugin>
                     <plugin>
-                        <groupId>${project.groupId}</groupId>
+                        <groupId>io.quarkus</groupId>
                         <artifactId>quarkus-maven-plugin</artifactId>
                         <version>${project.version}</version>
                         <executions>

